code/datums/ai/_ai_controller.dm

```diff
@@ -30,6 +30,10 @@ have ways of interacting with a specific atom and control it. They posses a blac
 	COOLDOWN_DECLARE(movement_cooldown)
 	///Delay between movements. This is on the controller so we can keep the movement datum singleton
 	var/movement_delay = 0.1 SECONDS
+	///A list for the path we're currently following, if we're using JPS pathing
+	var/list/movement_path
+	///Cooldown for JPS movement, how often we're allowed to try making a new path
+	COOLDOWN_DECLARE(repath_cooldown)
```

Comment:
I'm not a fan of all of these variables on the controller. I'll have to think of a better way of storing these variables but having a bunch of variables that are specific to movement datums is going to be detrimental overall

Reply:
I'm not sure where else you'd plan on putting them since the movement datums are singleton and iterate through the controllers, and the controllers are what's responsible for storing the pathing attempts, movement cooldown, and current movement target.

Reply:
maybe the blackboard? it'd be in line with the usage of it.